extensions/amp-viewer-integration/0.1/messaging.js

```diff
@@ -43,18 +44,34 @@ export let Message;
 
 export class WindowPortEmulator {
   /**
+   * @param {!Window} win
+   * @param {string} origin
+   */
+  constructor(win, origin) {
```

Comment:
Let's pass `target` here and use it instead of `win.parent`. Also, let's rename `origin` to `targetOrigin`. We will need this for `AmpViewerHost`.